cli/module_generate/scripts/generate_stubs.go

```diff
@@ -145,15 +171,31 @@ func parseFunctionSignature(resourceSubtype, resourceSubtypePascal string, funcD
 		return
 	}
 
+	// Receiver
+	receiver = modelType
+	if funcDecl.Recv != nil && len(funcDecl.Recv.List) > 0 {
+		field := funcDecl.Recv.List[0]
+		if starExpr, ok := field.Type.(*ast.StarExpr); ok {
+			if ident, ok := starExpr.X.(*ast.Ident); ok {
+				if ident.Name != "client" {
+					receiver = ident.Name
+				}
+			}
+		}
+	}
+
 	// Parameters
 	var params []string
 	if funcDecl.Type.Params != nil {
 		for _, param := range funcDecl.Type.Params.List {
 			paramType := formatType(param.Type)
-			if unicode.IsUpper(rune(paramType[0])) {
+			switch {
+			case unicode.IsUpper(rune(paramType[0])):
 				paramType = fmt.Sprintf("%s.%s", resourceSubtype, paramType)
-			} else if strings.HasPrefix(paramType, "[]") && unicode.IsUpper(rune(paramType[2])) {
+			case strings.HasPrefix(paramType, "[]") && unicode.IsUpper(rune(paramType[2])):
 				paramType = fmt.Sprintf("[]%s.%s", resourceSubtype, paramType[2:])
+			case strings.HasPrefix(paramType, "chan ") && unicode.IsUpper(rune(paramType[5])):
+				paramType = fmt.Sprintf("chan %s.%s", resourceSubtype, paramType[5:])
```

Comment:
(minor) this feels a little like magic to me. I believe it works but I have no sense of what `paramType[2]` or `paramType[5:]` means in the broader context, or why we care about them. Can we assign some descriptive variable names to these values so that if/when this code needs to be updated in the future, it's more clear what's happening here?

Reply:
`paramType` is the type of parameter. For example, it could be `"string"` and `"int"`. The indices are the characters of the string. So in a paramType string that looks like `"chan Tick"`, then `paramType[5]` is `T`.

Reply:
Basically this entire part of the code is checking to see if a parameter type (including in lists or if it's a `chan) starts with an uppercase letter. If so, add `<resource_name>.` in front of the type.